src/lib.rs

```diff
@@ -176,14 +177,22 @@ pub use poll::Poll;
 // Primitive futures
 mod done;
 mod empty;
+#[cfg(not(feature = "never"))]
 mod failed;
+#[cfg(not(feature = "never"))]
```

Comment:
Sure! We just need different names for the `never`-based `Failed` and `Finished`, as they conflict otherwise.